File: mRemoteV1/UI/UI.Window.ActiveDirectoryImport.vb
Imports mRemote3G.App
Imports WeifenLuo.WinFormsUI.Docking

Namespace UI

    Namespace Window
        Public Class ActiveDirectoryImport
            Inherits Base

#Region "Constructors"

            Public Sub New(panel As DockContent)
                InitializeComponent()

                Runtime.FontOverride(Me)

                WindowType = Type.ActiveDirectoryImport
                DockPnl = panel
            End Sub

#End Region

#Region "Private Methods"

#Region "Event Handlers"

            Private Sub ADImport_Load(sender As Object, e As EventArgs) Handles Me.Load
                ApplyLanguage()

                txtDomain.Text = ActiveDirectoryTree.Domain
                EnableDisableImportButton()
            End Sub

            Private Sub btnImport_Click(sender As Object, e As EventArgs) Handles btnImport.Click
                Import.ImportFromActiveDirectory(ActiveDirectoryTree.ADPath)
                DialogResult = DialogResult.OK
                Close()
            End Sub

            Private Shared Sub txtDomain_PreviewKeyDown(sender As Object, e As PreviewKeyDownEventArgs) _
                Handles txtDomain.PreviewKeyDown
                If e.KeyCode = Keys.Enter Then e.IsInputKey = True
            End Sub

            Private Sub txtDomain_KeyDown(sender As Object, e As KeyEventArgs) Handles txtDomain.KeyDown
                If e.KeyCode = Keys.Enter Then
                    ChangeDomain()
                    e.SuppressKeyPress = True
                End If
            End Sub

            Private Sub btnChangeDomain_Click(sender As Object, e As EventArgs) Handles btnChangeDomain.Click
                ChangeDomain()
            End Sub

            Private Sub ActiveDirectoryTree_ADPathChanged(sender As Object) Handles ActiveDirectoryTree.ADPathChanged
                EnableDisableImportButton()
            End Sub

#End Region

            Private Sub ApplyLanguage()
                btnImport.Text = Language.Language.strButtonImport
                lblDomain.Text = Language.Language.strLabelDomain
                btnChangeDomain.Text = Language.Language.strButtonChange
            End Sub

            Private Sub ChangeDomain()
                ActiveDirectoryTree.Domain = txtDomain.Text
                ActiveDirectoryTree.Refresh()
            End Sub

            Private Sub EnableDisableImportButton()
                btnImport.Enabled = Not String.IsNullOrEmpty(ActiveDirectoryTree.ADPath)
            End Sub

#End Region
        End Class
    End Namespace

End Namespace

File: mRemoteV1/CS/UI/UI.Window.ActiveDirectoryImport.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using AxWFICALib;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using AxMSTSCLib;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports

using WeifenLuo.WinFormsUI.Docking;
using mRemoteNG.App;


namespace mRemoteNG.UI.Window
{
	public partial class ActiveDirectoryImport : Base
			{
#region Constructors
				public ActiveDirectoryImport(DockContent panel)
				{
					InitializeComponent();
					
					Runtime.FontOverride(this);
					
					WindowType = Type.ActiveDirectoryImport;
					DockPnl = panel;
				}
#endregion
				
#region Private Methods
#region Event Handlers
				public void ADImport_Load(object sender, EventArgs e)
				{
					ApplyLanguage();
					
					txtDomain.Text = ActiveDirectoryTree.Domain;
					EnableDisableImportButton();
				}
				
				public void btnImport_Click(System.Object sender, EventArgs e)
				{
					Import.ImportFromActiveDirectory(ActiveDirectoryTree.ADPath);
					DialogResult = DialogResult.OK;
					Close();
				}
				
				static public void txtDomain_PreviewKeyDown(System.Object sender, PreviewKeyDownEventArgs e)
				{
					if (e.KeyCode == Keys.Enter)
					{
						e.IsInputKey = true;
					}
				}
				
				public void txtDomain_KeyDown(System.Object sender, KeyEventArgs e)
				{
					if (e.KeyCode == Keys.Enter)
					{
						ChangeDomain();
						e.SuppressKeyPress = true;
					}
				}
				
				public void btnChangeDomain_Click(System.Object sender, EventArgs e)
				{
					ChangeDomain();
				}
				
				public void ActiveDirectoryTree_ADPathChanged(object sender)
				{
					EnableDisableImportButton();
				}
#endregion
				
				private void ApplyLanguage()
				{
					btnImport.Text = My.Language.strButtonImport;
					lblDomain.Text = My.Language.strLabelDomain;
					btnChangeDomain.Text = My.Language.strButtonChange;
				}
				
				private void ChangeDomain()
				{
					ActiveDirectoryTree.Domain = txtDomain.Text;
					ActiveDirectoryTree.Refresh();
				}
				
				private void EnableDisableImportButton()
				{
					btnImport.Enabled = !string.IsNullOrEmpty(ActiveDirectoryTree.ADPath);
				}
#endregion
			}
}
